### PR TITLE
Comment the meaning of some pieces of the code and Java annotations

### PR DESCRIPTION
**Pull request by siluf**

### CODE REVIEW

File: src/main/java/es/unizar/webeng/hello/Application.java [hunk 1]
```diff
@@ -8,6 +8,9 @@
 @SpringBootApplication
 public class Application extends SpringBootServletInitializer {
 
+	/*
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/Application.java]: This is not a valid Javadoc block comment.

File: src/main/java/es/unizar/webeng/hello/HelloController.java [hunk 2]
```diff
@@ -20,7 +20,7 @@
      * If the key is not defined, the default value "Hello World" will be injected instead.
      */
     @Value("${app.message:Hello World}")
-    private String message;
+    private String message;						//message to display.
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/HelloController.java]: This comment should be part of the above comment

File: src/main/java/es/unizar/webeng/hello/HelloController.java [hunk 3]
```diff
@@ -36,7 +36,9 @@
      */
     @GetMapping("/")
     public String welcome(Map<String, Object> model) {
+		//Assign in the key "time" a new date.
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/HelloController.java]: Wrong alignment.

File: src/main/java/es/unizar/webeng/hello/HelloController.java [hunk 4]
```diff
@@ -36,7 +36,9 @@
...
         model.put("time", new Date());
+		//Assign in the key "message" the value of the message variable.
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/HelloController.java]: Wrong alignment.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 5]
```diff
@@ -12,14 +12,18 @@
...
 
-@RunWith(SpringRunner.class)
-@WebMvcTest(HelloController.class)
+
+@RunWith(SpringRunner.class)			//JUnit will invoke the SpringRunner class to run the tests in that class instead of the runner built into JUnit.
+@WebMvcTest(HelloController.class)		// Using this annotation will disable full auto-configuration and instead apply only configuration relevant to MVC tests.
+/*
+ * This class check if the message received from the client is correct.
+ */
 public class HelloControllerUnitTest {
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: This is not a valid javadoc comment. It must begin with `/**`

File: src/test/java/es/unizar/webeng/hello/IntegrationTest.java [hunk 6]
```diff
@@ -18,6 +18,9 @@
 @RunWith(SpringRunner.class)
 @SpringBootTest(webEnvironment = RANDOM_PORT)
 @DirtiesContext
+/*
+ * This class check if the messages that are sent to the server contains what is expected.
+ */
 public class IntegrationTest {
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/IntegrationTest.java]: Not a valid javadoc comment.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java (hunk 5 above)
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Pull request #8 is doing nearly the same.

File: src/main/java/es/unizar/webeng/hello/Application.java [hunk 7]
```diff
@@ -8,9 +8,9 @@
 @SpringBootApplication
 public class Application extends SpringBootServletInitializer {
 
-	/*
-	 *This main method uses SprinApplication.run() to launch the application.
-	 */
+   /**
+    *This main method uses SprinApplication.run() to launch the application.
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/Application.java]: There is a typo. In addition you should use `{@code ... }` for code.

File: src/main/java/es/unizar/webeng/hello/Application.java [hunk 8]
```diff
@@ -8,6 +8,9 @@
 @SpringBootApplication
 public class Application extends SpringBootServletInitializer {
 
+   /**
+    *This main method uses SprinApplication.run() to launch the application.
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/Application.java]: Add a whitespace after *

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 9]
```diff
@@ -12,14 +12,16 @@
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.assertThat;
 
-@RunWith(SpringRunner.class)
-@WebMvcTest(HelloController.class)
+
+@RunWith(SpringRunner.class)			//JUnit will invoke the SpringRunner class to run the tests in that class instead of the runner built into JUnit.
+@WebMvcTest(HelloController.class)		// Using this annotation will disable full auto-configuration and instead apply only configuration relevant to MVC tests.
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Use a javadoc comment.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 10]
```diff
@@ -12,14 +12,16 @@
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.assertThat;
 
-@RunWith(SpringRunner.class)
-@WebMvcTest(HelloController.class)
+
+@RunWith(SpringRunner.class)			//JUnit will invoke the SpringRunner class to run the tests in that class instead of the runner built into JUnit.
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Use a javadoc comment.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 11]
```diff
@@ -12,14 +12,16 @@
...
+@WebMvcTest(HelloController.class)		// Using this annotation will disable full auto-configuration and instead apply only configuration relevant to MVC tests.
+
 public class HelloControllerUnitTest {
 
-    @Value("${app.message:Hello World}")
+    @Value("${app.message:Hello World}")		// Annotation that indicates a default value expression for the affected argument.
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Use a javadoc comment.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 12]
```diff
@@ -12,14 +12,16 @@
...
     private String message;
 
-    @Autowired
+    @Autowired									// Annotation to auto wire bean on the field.
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Use a javadoc comment.

**Comment by fjlopez** [comment]
Resolve the conflicts caused by pull #8 with your code.

File: src/main/java/es/unizar/webeng/hello/Application.java [hunk 13]
```diff
@@ -8,6 +8,9 @@
 @SpringBootApplication
 public class Application extends SpringBootServletInitializer {
 
+   /**
+    * This main method uses {@code SprinApplication.run(Application.class, args)} to launch the application.
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/Application.java]: Check the spelling. It is `SpringApplication`, not `SprinApplication`.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 14]
```diff
@@ -13,6 +13,23 @@
 import static org.junit.Assert.assertThat;
 
 /**
+*   Annotations used in the code:<br>
+*   <em>@RunWith(SpringRunner.class)</em>  Annotation from Junit, Junit will run using Spring's testing support.
+*	 JUnit will invoke the SpringRunner class to run the tests in that class instead of the runner built into JUnit.
+*	 SpringRunner is the new name for SpringJUnit4ClassRunner.<br>
+*	 External link for more info <a href="http://junit.sourceforge.net/javadoc/org/junit/runner/RunWith.html">sourceforge.net RunWith</a><br><br>
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Ideal `@see` annotation. e.g.
```
@see <a href="http://junit.org/junit4/javadoc/latest/org/junit/runner/RunWith.html">org.junit.runner.RunWith</a>
```
Note that I point to the latest version.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 15]
```diff
@@ -13,6 +13,23 @@
...
+*
+*   <em>@WebMvcTest(HelloController.class)</em>  Start the Spring application context without the server, 
+*	in that case Spring Boot is only instantiating one controller HelloController.
+*	Using this annotation will disable full auto-configuration and instead apply only configuration relevant to MVC tests.<br>
+*	External link for more info <a href="https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/autoconfigure/web/servlet/WebMvcTest.html">spring.io WebMvcTest</a><br><br>
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Same as above.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 16]
```diff
@@ -13,6 +13,23 @@
...
+*
+*   <em>@Value("${app.message:Hello World}")</em> Annotation that indicates a default value expression for the affected argument.
+*   We can inject the value from a file into a variable with that syntax  @Value("${value.from.file}"), in case it's not defined 
+*	we use double dots @Value("${unknown.param:Hello World}") and the text Hello World will be injected.<br>
+*	External link for more info <a href="https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Value.html">spring.io Value</a><br><br>
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Same as above.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java (hunk 14 above)
Comment (siluf) [reply src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: I'm going to correct it but that has been written by #8 . I've just added an extra line in each annotation.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 17]
```diff
@@ -13,23 +13,27 @@
...
-*	External link for more info <a href="https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Value.html">spring.io Value</a><br><br>
+*	
 *
-*   <em>@Autowired</em> This annotation will scan on the packpage for annotations like Controller,
+*   <em>@Autowired</em> Annotation to auto wire bean on the field. This annotation will scan on the packpage for annotations like Controller,
 *	 Component, Repository or simple bean, to auto-inject all the necesary into the variable.<br>
-*	External link for more info <a href="https://docs.spring.io/spring-framework/docs/2.5.x/api/org/springframework/beans/factory/annotation/Autowired.html">spring.io Autowired</a><br><br>
+*	
+*   @see <a href="https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Value.html">org.springframework.beans.factory.annotation.Value</a>
+*   @see <a href="https://docs.spring.io/spring-framework/docs/2.5.x/api/org/springframework/beans/factory/annotation/Autowired.html">org.springframework.beans.factory.annotation.Autowired</a>
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Fix my example by pointing to the current documentation.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 18]
```diff
@@ -13,23 +13,27 @@
...
 *	 SpringRunner is the new name for SpringJUnit4ClassRunner.<br>
-*	External link for more info <a href="http://junit.sourceforge.net/javadoc/org/junit/runner/RunWith.html">sourceforge.net RunWith</a><br><br>
+*	 External link for more info @see <a href="http://junit.sourceforge.net/javadoc/org/junit/runner/RunWith.html">sourceforge.net RunWith</a><br><br>
 *
 *   <em>@WebMvcTest(HelloController.class)</em>  Start the Spring application context without the server, 
-*	in that case Spring Boot is only instantiating one controller HelloController.<br>
-*	External link for more info <a href="https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/autoconfigure/web/servlet/WebMvcTest.html">spring.io WebMvcTest</a><br><br>
+*	in that case Spring Boot is only instantiating one controller HelloController.
+*	Using this annotation will disable full auto-configuration and instead apply only configuration relevant to MVC tests.<br>
+*	External link for more info @see  <a href="https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/autoconfigure/web/servlet/WebMvcTest.html">spring.io WebMvcTest</a><br><br>
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Add to the list of @see below.

File: src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java [hunk 19]
```diff
@@ -13,23 +13,27 @@
 import static org.junit.Assert.assertThat;
 
 /**
-*    Annotations used in the code:<br>
+*   Annotations used in the code:<br>
 *   <em>@RunWith(SpringRunner.class)</em>  Annotation from Junit, Junit will run using Spring's testing support.
+*	 JUnit will invoke the SpringRunner class to run the tests in that class instead of the runner built into JUnit.
 *	 SpringRunner is the new name for SpringJUnit4ClassRunner.<br>
-*	External link for more info <a href="http://junit.sourceforge.net/javadoc/org/junit/runner/RunWith.html">sourceforge.net RunWith</a><br><br>
+*	 External link for more info @see <a href="http://junit.sourceforge.net/javadoc/org/junit/runner/RunWith.html">sourceforge.net RunWith</a><br><br>
```
Comment (fjlopez) [inline src/test/java/es/unizar/webeng/hello/HelloControllerUnitTest.java]: Add to the list of @see below

**Pull request merged**